src/sagemaker/jumpstart/types.py

```diff
@@ -944,6 +1003,273 @@ def to_json(self) -> Dict[str, Any]:
                     json_obj[att] = cur_val
         return json_obj
 
+
+class JumpStartPresetComponent(JumpStartMetadataBaseFields):
```

Comment:
later we should refactor this file, it's wayyy too big

Reply:
yeah, I can move some types to a separate file